fix(plugin): Fix ability to have custom value for openebs.io/nodeid

jnels124 opened the pull request:
## Pull Request template

Please, go through these steps before you submit a PR.

**Why is this PR required? What issue does it fix?**:
This PR is required in order to allow a ZFS volume to be moved to another node
**What this PR does?**:
This pr fixes existing logic that attempts to retrieve a node using the name and instead retrieves the node using the nodeid label
**Does this PR require any upgrade changes?**:
No
**If the changes in this PR are manually verified, list down the scenarios covered:**:
These changes were manually verified by doing the following:

1. Labeled all nodes via `kubectl label node <Node> openebs.io/nodeid=value-1`
2. Created persistent volumes
3. Ensured that the ZFSNode resources were created correctly.
4. Performed manual upgrade of the cluster
5. Attached the label with distinct value to each of the new nodes
6. Started test application to ensure that the volumes were mounted and accessible

**Any additional information for your reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_


**Checklist:**
- [x] Fixes #450
- [x] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [x] Has the change log section been updated?
- [ ] Commit has unit tests
- [ ] Commit has integration tests
- [x] (Optional) Are upgrade changes included in this PR? If not, mention the issue/PR to track:
- [x] (Optional) If documentation changes are required, which issue on https://github.com/openebs/openebs-docs is used to track them:

jnels124 committed:
fix(plugin): fix node id to node mapping

Signed-off-by: Jesse Nelson <jesse@swirldslabs.com>

jnels124 commented on pkg/mgmt/zfsnode/start.go:
```diff
@@ -60,10 +64,26 @@ func Start(controllerMtx *sync.RWMutex, stopCh <-chan struct{}) error {
+	k8sNodeCandidates, err := kubeClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{
+		LabelSelector: topologySelector,
+	})
+
```
Update log below

jnels124 committed:
add change log

Signed-off-by: Jesse Nelson <jesse@swirldslabs.com>

jnels124 committed:
fix case when nodeid not set

Signed-off-by: Jesse Nelson <jesse@swirldslabs.com>

jnels124 committed:
simplify logic

Signed-off-by: Jesse Nelson <jesse@swirldslabs.com>

jnels124 commented:
@niladrih when you have a moment would you mind taking a peak at this PR. It is a critical fix in order for our deployment. Without this change, we will not be able to perform node upgrades within the cluster.

sinhaashish commented:
@jnels124  sorry to keep you waiting, but @niladrih is having some higher priority jobs at the moment and is stuck in them . For sure i will look into this PR. Just give us some more time.

steven-sheehy commented on pkg/driver/controller.go:
Should have some tests for the changes to both go classes

hrudaya21 commented:
@jnels124 Can you please rebase your code. Also please add the testing steps in the Documentation file as part of this PR.

jnels124 committed:
update doc

Signed-off-by: Jesse Nelson <jesse@swirldslabs.com>

jnels124 commented:
@hrudaya21 I have rebased and updated install documentation to reference existing documentation. Please let me know if anything else is needed to get this merged.

jnels124 committed:
fix cases where nodeid not set

Signed-off-by: Jesse Nelson <jesse@swirldslabs.com>

jnels124 committed:
check for error first

Signed-off-by: Jesse Nelson <jesse@swirldslabs.com>

DeprecatedLuke commented:
See #489. Might still be broken.
